testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentUp.java

```diff
@@ -90,6 +92,9 @@ public Integer call()
         @Option(names = "--environment", paramLabel = "<environment>", description = "Name of the environment to start", required = true)
         public String environment;
 
+        @Option(names = "--option", paramLabel = "<option>", description = "Extra option (property can be used multiple times; format is key=value)")
+        public final List<ExtraOption> extraOptions = new ArrayList<>();
```

Comment:
Why introducing custom type? picocli handles maps out of the box: https://picocli.info/#_maps